lib/fbe/middleware/logging_formatter.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+# MIT License
+#
+# Copyright (c) 2024 Zerocracy
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+require 'faraday'
+
+# Faraday logging formatter show verbose log for only error response
+class Fbe::Middleware::LoggingFormatter < Faraday::Logging::Formatter
+  AUTHORIZATION_FILTER = [/(Authorization: )([^&]+)([^&]{5})/, '\1********\3'].freeze
+
+  def initialize(**)
+    super
+    filter(*AUTHORIZATION_FILTER)
+  end
+
+  def request(env)
+    super unless log_only_errors?
+  end
+
+  def response(env)
+    return super unless log_only_errors?
+    request_with_response(env) if error_status?(env)
+  end
+
+  def request_with_response(env)
+    old_log_level = @options[:log_level]
+    @options[:log_level] = :error
+    public_send(log_level, 'request') do
```

Comment:
@Suban05 this piece of code borrowed from [here](https://github.com/lostisland/faraday/blob/59c5003ceb350096ade65086f8c17efbb7e0e53c/lib/faraday/logging/formatter.rb#L25-L39) .
I changed only `env[:body]` to `env[:request_body]` and `env[:response_body]` since the response of the body was duplicated.
`public_send` - can call only public method, `send` - can call also private method, I think to use `public_send` justified.